build.gradle

```diff
@@ -148,26 +149,55 @@ tasks.register('checkAkitInstall', JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
 }
 
-tasks.register('installLocalGitHookPostCheckout', Copy) {
-    from new File(rootProject.rootDir, '.scripts/post-checkout')
-    into { new File(rootProject.rootDir, '.git/hooks') }
-    filePermissions {
-        unix(0775)
-    }
+// Setup local git settings
+def gitConfig = 'Git Configuration'
+tasks.register('installLocalGitHookSymlinks') {
+    group = gitConfig
+
+    outputs.dir "$rootDir/.git/hooks/"
+
     doLast {
-        println("---Added Post Checkout Git Hook Script---")
+        def linkDir = file("$rootDir/.git/hooks/").toPath()
+        def gitHooks = fileTree("$rootDir/.scripts/git-hooks/")
+        gitHooks.forEach { File target ->
+            def link = linkDir.resolve(target.toPath().getFileName())
+
+            if (!Files.isSymbolicLink(link)) {
+                if (Files.exists(link)) {
+                    Files.delete(link)
+                }
+
+                def rel = { f -> rootDir.toPath().relativize(f) }
+                Files.createSymbolicLink(link, target.toPath())
+
+
+                println("---Installed Symbolic Link ${rel(link)} -> ${rel(target.toPath())} ---")
+            }
+        }
     }
 }
 
-tasks.register('installLocalGitHookPreCommit', Copy) {
-    from new File(rootProject.rootDir, '.scripts/pre-commit')
-    into { new File(rootProject.rootDir, '.git/hooks') }
-    filePermissions {
-        unix(0775)
-    }
-    doLast {
-        println("---Added Pre Commit Git Hook Script---")
-    }
+tasks.register('configLocalGitBlameIgnoreRevsFile', Exec) {
+    group = gitConfig
+
+    workingDir rootProject.rootDir
+    commandLine 'git', 'config', '--local', 'blame.ignoreRevsFile', '.git-blame-ignore-revs'
+
+    outputs.file('.git/config')
+}
+
+tasks
+        .matching { it.group != gitConfig }
+        .configureEach {
+            it.dependsOn tasks.matching {
+                it.group == gitConfig
+            }
+        }
+
+if (System.getenv().containsKey('CI') || !file('.git').isDirectory()) {
+    tasks
+            .matching { it.group == gitConfig }
+            .configureEach { it.enabled = false }
```

Comment:
maybe make it one line, also when does it run?

Reply:
When running in CI or where there is no git directory

Reply:
i meant not the inside when it is actually calling the if?